release-notes.txt

```diff
@@ -1,3 +1,548 @@
+* boards/nucleo-c031c6: fix ADC and enhance doc (#20502)
+* doc/roadmap: Make working 6LBR a goal (#20351)
+* doc: Allow resizing of doxygen side-nav (#20511)
+
+Build System / Tooling
+----------------------
+
++ build sytem: add `--no-warn-rwx-segments` to linker (#20381)
++ dist/tools/mspdebug: build from source (#20345)
+* Avoid using egrep (#20522)
+* build system: Improve failure mode for FEAUTRES_REQUIRED_ANY (#20408)
+* dist/tools/dhcpv6-pd_ia: recognize Debian correctly (#20487)
+* Enable compile_and_test_for_board to skip if nothing changed (#19546)
+* tools/openocd: define SWD_EN signal for tigard (#20507)
+* tree wide: accomodate RISC-V toolchain update (#20380)
+
+Kconfig
+-------
+
+* Remove Kconfig dependency modelling (#20438)
+
+Examples
+--------
+
+* examples/filesystem: Update README and increase default stack size (#20439)
+* examples/psa_crypto: Fix modules to run ECDSA on SEs (#20461)
+* examples/wakaama: rename to lwm2m (#20327)
+* exmple > example in Cargo.toml comment (#20399)
+
+Testing
+-------
+
++ tests/shell: add test for preemption (#19005)
++ tests: Add cases to the scn_u32_hex unittest (#20463)
+* ci: disable some more flakey native tests (#20358)
+* examples, tests: disable native and native64 for more flaky tests (#20401)
+* make use of nanocoap_server_auto_init (#20378)
+* Refactor tests periph uart (#20281)
+* release-test.yml: return of the Matrix notification (#20424)
+* tests/pkg/lwip: include default-radio-settings.inc.mk (#20325)
+* tests/sys/shell: increase the stack size of periodic thread (#20548)
+* tests/sys/timer_overhead: disable test on native in CI (#20324)
+* tests/unittests/README: link to Supported Boards updated (wiki ->
+  site) (#20500)
+* tests: 64-bit support preparations (#20276)
+* tests: Cover API of scn_u32_hex in the unittest (#20458)
+* tests: disable flaky tests on `native` in the CI (#20382)
+* unittests: Move at test to driver tests (#20540)
+
+API Changes
+-----------
+
++ drivers/at: fix URC handling and add better testing (#20423)
+* drivers/at: parse +CME/+CMS responses and save error value (#20289)
+* gcoap: make gcoap_req_send_tl() an alias of gcoap_req_send() (#20514)
+* gcoap_fileserver: rename to nanocoap_fileserver (#20030)
+* makefiles, treewide: Remove MCU variable (#20397)
+
+Uncategorized
+-------------
++ rdm/README: add rdm0003 to the list (#20469)
++ release-notes.txt: add 2024.01 release notes (#20350)
++ Roadmap: Add Home-Assistant integration on the roadmap (#20403)
++ vscode: Add hard-coded path to compile_commands.json (#20417)
+* RDM: Naming boards (#18589)
+* README: Change Section titles from all caps to regular (#20405)
+
+And 59 minor changes.
+
+Deprecations
+============
+
+Deprecations (3)
+----------------
+
+* gnrc_mac / gnrc_lwmac / gnrc_gomach: Deprecate modules (#20512)
+* gnrc_netif: remove deprecated CONFIG_GNRC_NETIF_LORAWAN_NETIF_HDR (#20515)
+* sys/phydat: Remove deprecated print function (#20529)
```

Comment:
```suggestion
* Deprecate MCU preprocessor defines (#20397)
```

We've had that before, but it's a build system change and a prepocessor change rolled into one

Reply:
Huh? The removal and addition do not seem related?

Reply:
Sorry, should have been an addition rather than a replacement; edited.